lib/cn/kblock_reader.c

```diff
@@ -194,70 +128,54 @@ kbr_read_metrics(struct kvs_mblk_desc *kblkdesc, struct kblk_metrics *metrics)
     return 0;
 }
 
-static merr_t
-kbr_madvise_region(struct kvs_mblk_desc *kblkdesc, u32 pg, u32 pg_cnt, int advice)
-{
-    u32 pg_max = pg + pg_cnt;
-
-    while (pg < pg_max) {
-        u32 chunk = min_t(u32, pg_max - pg, HSE_RA_PAGES_MAX);
-        merr_t err;
-
-        err = mpool_mcache_madvise(
-            kblkdesc->map, kblkdesc->map_idx, PAGE_SIZE * pg, PAGE_SIZE * chunk, advice);
-        if (ev(err))
-            return err;
-
-        pg += chunk;
-    }
-
-    return 0;
-}
-
 void
-kbr_madvise_kmd(struct kvs_mblk_desc *kblkdesc, struct wbt_desc *desc, int advice)
+kbr_madvise_kmd(struct kvs_mblk_desc *md, struct wbt_desc *wbd, int advice)
 {
     merr_t err;
-    u32    pg = desc->wbd_first_page + desc->wbd_root + 1;
-    u32    pg_cnt = desc->wbd_kmd_pgc;
+    uint32_t pg = wbd->wbd_first_page + wbd->wbd_root + 1;
+    uint32_t pg_cnt = wbd->wbd_kmd_pgc;
 
-    err = kbr_madvise_region(kblkdesc, pg, pg_cnt, advice);
-
-    ev(err);
+    if (pg_cnt) {
+        err = mblk_madvise_pages(md, pg, pg_cnt, advice);
+        ev(err);
+    }
 }
 
 void
-kbr_madvise_wbt_leaf_nodes(struct kvs_mblk_desc *kblkdesc, struct wbt_desc *desc, int advice)
+kbr_madvise_wbt_leaf_nodes(struct kvs_mblk_desc *md, struct wbt_desc *wbd, int advice)
 {
     merr_t err;
-    u32    pg = desc->wbd_first_page;
-    u32    pg_cnt = desc->wbd_leaf_cnt;
-
-    err = kbr_madvise_region(kblkdesc, pg, pg_cnt, advice);
+    uint32_t pg = wbd->wbd_first_page;
```

Comment:
`const`, also it would be nice to have consistent variable names

Reply:
Var names are consistent.  In these 4 madvise functions local vars are `err`, `pg` and `pg_cnt`.

Reply:
In a previous function, it was `pg` and `pgc` whereas these are what you describe.

Reply:
I try to minimize non-functional code changes in large PRs like this.  It helps me keep track of my changes, reduces merge conflicts, results in fewer lines of code to change.